File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int l = tokens.size();
        stack<int> st;
        for (int i = 0; i < l; i++) {
            if (tokens[i] != "+" && tokens[i] != "-" && tokens[i] != "*" && tokens[i] != "/") {
                st.push(stoi(tokens[i])); 
            } else {
                int b = st.top();
                st.pop();
                int a = st.top();
                st.pop();
                int r;
                if (tokens[i] == "+") {
                    r = a + b;
                } else if (tokens[i] == "-") {
                    r = a - b;
                } else if (tokens[i] == "*") {
                    r = a * b;
                } else if (tokens[i] == "/") {
                    r = a / b;
                }
                st.push(r);
            }
        }
        int result = st.top();
        st.pop();
        return result;
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();
        
        for(String s : tokens){
            
            // System.out.println(s + "str");
            
            if(!s.equals("+") && !s.equals("-") && !s.equals("*") && !s.equals("/")){
                int val = Integer.valueOf(s);
                // System.out.println(val + "val");
                st.push(val);
            }
            
            else{
                int num2 = st.pop();
                int nums1 = st.pop();
                int res = 0;
                
                if(s.equals("+")){
                    res = nums1 + num2;
                }
                else if(s.equals("-")){
                    res = nums1 - num2;
                }
                else if(s.equals("*")){
                    res = nums1 * num2;
                }
                else if(s.equals("/")){
                    res = nums1 / num2;
                }
                
                // System.out.println(res + "res");
                st.push(res);
            }
        }
        
        return st.peek();
    }
}